cwltest/__init__.py

```diff
@@ -470,9 +470,23 @@ def main():  # type: () -> int
 
     for t in tests:
         if t.get("label"):
+            _logger.warning(
+                "The `label` field is deprecated. Use `id` field instead. "
+                "It will become an error since 2023/1/1."
```

Comment:
>It will become an error since 2023/1/1.

Normally I don't see a specific date in deprecation warnings. Maybe that could go in the release notes/readme instead? I think we could also say "from 2023/1/1", or will be deprecated on Q1/2023, etc.

Reply:
I sometimes see such messages in dlang community ([example](https://github.com/dlang/phobos/pull/7840/files)) and find that it is useful for users as well as for developers.
Of course it is better to include it in release note or readme in addition to deprecation messages.

If it is in the message, users and developers can see the deprecation date as soon as they see the message.
Otherwise they have to search for the deprecation date in both of the release notes and readme because they do not know where it is or that there are no documents for it (in the worst case).

---
"2023/1/1", "Q1/2023", "when CWL v1.2.1 is released" and others are OK for me for the deprecation deadline.

Reply:
I removed the deadline messages.